ssas/service/public/oktalive_test.go

```diff
@@ -0,0 +1,184 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/CMSgov/bcda-app/ssas/okta"
+)
+
+type OktaLiveTestSuite struct {
+	suite.Suite
+	oc *OktaMFAPlugin
+	email string
+	userId string
+	smsFactorId string
+}
+
+func (s *OktaLiveTestSuite) SetupSuite() {
+	s.email = os.Getenv("OKTA_MFA_EMAIL")
+	s.userId = os.Getenv("OKTA_MFA_USER_ID")
+	s.smsFactorId = os.Getenv("OKTA_MFA_SMS_FACTOR_ID")
+
+	if s.email == "" || s.userId == "" || s.smsFactorId == "" {
+		s.FailNow(fmt.Sprintf("Cannot run live Okta tests without env vars set: OKTA_MFA_EMAIL=%s; OKTA_MFA_USER_ID=%s; OKTA_MFA_SMS_FACTOR_ID=%s", s.email, s.userId, s.smsFactorId))
+	}
+}
+
+func (s *OktaLiveTestSuite) SetupTest() {
+	s.oc = NewOktaMFA(okta.Client())
+}
+
+func (s *OktaLiveTestSuite) TestPostFactorChallengeSuccess() {
+	trackingId := uuid.NewRandom().String()
+	factor := Factor{Id: s.smsFactorId, Type: "sms"}
+
+	factorVerification, err := s.oc.postFactorChallenge(s.userId, factor, trackingId)
+	if err != nil || factorVerification == nil {
+		s.FailNow("factor result not parsed: " + err.Error())
+	}
+	assert.Equal(s.T(), "CHALLENGE", factorVerification.Result)
+
+	// A second SMS request within 30 seconds will fail.  This second test case must
+	// follow in the same unit test as the successful test case.
+	factorVerification, err = s.oc.postFactorChallenge(s.userId, factor, trackingId)
+	if err == nil || factorVerification != nil {
+		s.FailNow("second request should not be successful")
+	}
+}
+
+func (s *OktaLiveTestSuite) TestPostFactorChallengeInvalidFactor() {
+	trackingId := uuid.NewRandom().String()
+	factor := Factor{Id: "abcdefg1234567", Type: "sms"}
+
+	factorVerification, err := s.oc.postFactorChallenge(s.userId, factor, trackingId)
+	if err == nil || factorVerification != nil {
+		s.FailNow("invalid factor should not be successful")
+	}
+}
+
+func (s *OktaLiveTestSuite) TestPostFactorChallengeInvalidUser() {
+	trackingId := uuid.NewRandom().String()
+	userId := "abcdefg1234567"
+	factor := Factor{Id: s.smsFactorId, Type: "sms"}
+
+	factorVerification, err := s.oc.postFactorChallenge(userId, factor, trackingId)
+	if err == nil || factorVerification != nil {
+		s.FailNow("invalid factor should not be successful")
+	}
+}
+
+func (s *OktaLiveTestSuite) TestPinnedKeyNotMatched() {
+	originalOktaCACertFingerprint := okta.OktaCACertFingerprint
+	okta.OktaCACertFingerprint, _ = hex.DecodeString("00112233aabbcc")
+	s.oc = NewOktaMFA(okta.Client())
+
+	trackingId := uuid.NewRandom().String()
+	factor := Factor{Id: s.smsFactorId, Type: "sms"}
+
+	factorVerification, err := s.oc.postFactorChallenge(s.userId, factor, trackingId)
+	if err == nil || factorVerification != nil {
+		s.FailNow("certificate signed by unexpected CA should abort TLS handshake")
+	}
+	okta.OktaCACertFingerprint = originalOktaCACertFingerprint
+}
+
+func (s *OktaLiveTestSuite) TestGetUserSuccess() {
+	fmt.Println("s.userId", s.userId)
+	fmt.Println("s.email", s.email)
```

Comment:
I suspect these were not intentionally left in :) (Or maybe they were!)

Reply:
Good catch!